### PR TITLE
005 system email approval

### PR DESCRIPTION
**Pull request by eyalgolan**
Completed:
* email verification & user activation feature
* email DB connection error feature

### CODE REVIEW

File: src/server/server.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import User from './src/models/user'
 import mailgun from 'mailgun-js' 
+import { generateDbConnectionFailedEmail } from './src/resources/emails'
+
 const path = require('path');
```
Comment (arielfenster) [inline src/server/server.js:5]: change 'require' to 'import'

File: src/server/src/routes/users.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import bcrypt from 'bcryptjs'
 const { valid } = require('@hapi/joi');
 import mailgun from 'mailgun-js'
 import jwt from 'jsonwebtoken'
-import generateRegistrationEmail from '../resources/emails'
+import { generateRegistrationEmail } from '../resources/emails'
 require('dotenv').config();
```
Comment (arielfenster) [inline src/server/src/routes/users.js:9]: require -> import

File: src/client/src/App.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { BrowserRouter as Router, Route } from "react-router-dom";
 import Navbar from './routes/navbar';
 import CreateUser from './routes/CreateUserPage/components/create-user';
 import LoginUser from './routes/LoginUserPage/components/login-user';
+import ActivateUser from './routes/ActivateUserPage/components/activate-user'
```
Comment (arielfenster) [inline src/client/src/App.js:7]: component file names

File: src/client/src/routes/ActivateUserPage/components/activate-user.js [hunk 4]
```diff
@@ -1,15 +1,31 @@
...
-function ActivateUser(props) {
-    try {
-        console.log(window.location.href);
-        //const res = await axios.post('http://localhost:5000/users/activation', user);
-    }
-    catch (err) {
-        console.log(err);
+function ActivateUser (props) {
+    const onActivate = async (e) => {
+        try {
```
Comment (arielfenster) [inline src/client/src/routes/ActivateUserPage/components/activate-user.js:6]: why nested try blocks?

File: src/server/src/routes/users.js [hunk 5]
```diff
@@ -91,17 +91,23 @@ router.post('/login', async (req,res) => {
...
-    jwt.verify(token, process.env.JWT_ACC_ACTIVATE, function(err, decodeToken) {
+    const res = await jwt.verify(token, process.env.JWT_ACC_ACTIVATE, function(err, decodeToken) {
       if(err) {
+        console.log(err);
         return res.status(400).json({error: 'Incorrect or expired link'})
       }
+      else{
+        console.log("Activation success");
+      }
       const {name, email, password} = decodeToken;
```
Comment (arielfenster) [inline src/server/src/routes/users.js:105]: what is the usage of these variables?

File: src/server/server.js (hunk 1 above)
Comment (eyalgolan) [reply src/server/server.js]: fixed

File: src/server/src/routes/users.js (hunk 2 above)
Comment (eyalgolan) [reply src/server/src/routes/users.js]: fixed

File: src/client/src/App.js (hunk 3 above)
Comment (eyalgolan) [reply src/client/src/App.js]: fixed

File: src/client/src/routes/ActivateUserPage/components/activate-user.js (hunk 4 above)
Comment (eyalgolan) [reply src/client/src/routes/ActivateUserPage/components/activate-user.js]: You're right, broke it down to two separate functions

File: src/server/src/routes/users.js (hunk 5 above)
Comment (eyalgolan) [reply src/server/src/routes/users.js:107]: I'll remove them

**Comment by arielfenster** [comment]
already pushed to dev

**Pull request closed without merging**